src/plugins/Sharesite/Freesite.java

```diff
@@ -90,6 +111,35 @@ public synchronized void setName(String name) {
 		this.name = name;
 	}
 
+	public synchronized String getPath() {
+		if (path != null) {
+			return path;
+		} else {
+			return name;
+		}
+	}
+
+	public synchronized void setPath(String path) {
+		this.path = path;
+	}
+
+	public synchronized Integer getInsertHour() {
+		if (insertHour != null) {
+			return insertHour;
+		} else {
+			return -1; // meaning unrestricted
+		}
+	}
+
+	public synchronized void setInsertHour(Integer insertHour) {
+		if (insertHour.equals(-2)) {
```

Comment:
how can I make null parse to -2? (that would be the best action: in case of error choose the safe option)